src/EventListener/InitializeSystemListener.php

```diff
@@ -181,10 +190,7 @@ protected function boot(Request $request = null)
         $this->initializeLegacySessionAccess();
         $this->setDefaultLanguage();
 
-        // Fully load the configuration
-        $objConfig = Config::getInstance();
```

Comment:
Here we should call `$this->config->instantiate()`. The adapter must not instantiate the class magically.